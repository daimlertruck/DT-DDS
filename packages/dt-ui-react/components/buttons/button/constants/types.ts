import { ComponentSize } from '../../../../types/componentSize';

export type ButtonVariant = 'solid' | 'outlined' | 'text';

export type ButtonColor = 'primary' | 'error' | 'neutral';

export type ButtonSize = Extract<ComponentSize, 'medium' | 'large' | 'small'>;
