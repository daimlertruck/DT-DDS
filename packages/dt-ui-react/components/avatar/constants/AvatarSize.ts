export const AvatarSize = {
  Small: 'small',
  Medium: 'medium',
  Large: 'large',
} as const;

export type AvatarSize = (typeof AvatarSize)[keyof typeof AvatarSize];
