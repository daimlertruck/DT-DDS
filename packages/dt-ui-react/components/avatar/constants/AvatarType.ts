export enum AvatarType {
  Profile = 'profile',
  TeamMember = 'teamMember',
}
