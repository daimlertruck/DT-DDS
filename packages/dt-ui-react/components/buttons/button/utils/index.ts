export * from './buttonVariantColors';
export * from './buttonVariantStyles';
