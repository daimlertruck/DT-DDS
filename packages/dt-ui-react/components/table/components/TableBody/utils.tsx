import { Children, ReactNode, cloneElement, isValidElement } from 'react';

import { TableDataCellHeader } from '../TableDataCell/TableDataCell';

export const addHeadersToDataCells = (
  dataCells: ReactNode,
  headers: ReactNode[]
) => {
  return Children.map(dataCells, (dataCell, index) => {
    if (isValidElement(dataCell)) {
      const header = headers[index];
      return cloneElement(
        dataCell,
        {},
        <>
          <TableDataCellHeader>{header}</TableDataCellHeader>
          {dataCell.props.children}
        </>
      );
    }
    return dataCell;
  });
};

export const addHeadersToDataCellsInRows = (
  rows: ReactNode,
  headers: ReactNode[]
) => {
  return Children.map(rows, (row) => {
    if (isValidElement(row)) {
      return cloneElement(
        row,
        {},
        addHeadersToDataCells(row.props.children, headers)
      );
    }
    return row;
  });
};
