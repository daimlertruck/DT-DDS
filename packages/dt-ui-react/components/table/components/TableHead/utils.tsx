import { Children, ReactNode, isValidElement } from 'react';

const extractColumnHeaders = (columnHeaders: ReactNode): ReactNode[] => {
  return Children.toArray(columnHeaders).reduce<ReactNode[]>(
    (headers, columnHeader) => {
      if (isValidElement(columnHeader)) {
        return [...headers, columnHeader.props.children];
      }

      return headers;
    },
    []
  );
};

export const extractTableHeaders = (theadRow: ReactNode) => {
  if (isValidElement(theadRow)) {
    return extractColumnHeaders(theadRow.props.children);
  }

  return [];
};
