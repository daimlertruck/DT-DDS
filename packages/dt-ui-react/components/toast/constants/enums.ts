export enum ToastPosition {
  TopLeft = 'top-left',
  TopCenter = 'top-center',
  TopRight = 'top-right',
  BottomLeft = 'bottom-left',
  BottomCenter = 'bottom-center',
  BottomRight = 'bottom-right',
}

export enum ToastType {
  Success = 'success',
  Error = 'error',
  Info = 'info',
  /* react-hot-toast types
  Loading = 'loading',
  Blank = 'blank',
  Custom = 'custom',
  */
}
