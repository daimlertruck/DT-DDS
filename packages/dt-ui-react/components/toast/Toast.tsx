import {
  InfoOutlineIcon,
  ErrorOutlineIcon,
  CheckCircleOutlineIcon,
  WarningIcon,
  CloseIcon,
} from '../../core/assets/svgs';
import { BaseProps } from '../../types';

import { ToastType } from './constants';
import {
  ToastActionsStyled,
  ToastButtonCloseStyled,
  ToastContentStyled,
  ToastIconStyled,
  ToastMessageStyled,
  ToastStyled,
  ToastTextContainer,
  ToastTitleStyled,
} from './Toast.styled';

export interface ToastProps extends BaseProps {
  id: string;
  type: ToastType;
  title: string;
  message: string;
  onClose: () => void;
  isVisible?: boolean;
  dismissible?: boolean;
}

const ToastIcons = {
  [ToastType.Success]: CheckCircleOutlineIcon,
  [ToastType.Error]: ErrorOutlineIcon,
  [ToastType.Info]: InfoOutlineIcon,
  [ToastType.Warning]: WarningIcon,
};

const Toast = ({
  children,
  id,
  dataTestId,
  title,
  message,
  onClose,
  type,
  isVisible = true,
  dismissible = true,
}: ToastProps) => {
  const Icon = ToastIcons[type];
  const dataTest = dataTestId ?? `toast-${id}`;

  return (
    <ToastStyled
      data-testid={dataTest}
      key={id}
      type={type}
      isVisible={isVisible}
    >
      <ToastIconStyled type={type}>
        <Icon />
      </ToastIconStyled>
      <ToastContentStyled>
        <ToastTextContainer hasCloseButton={dismissible}>
          <ToastTitleStyled>{title}</ToastTitleStyled>
          {dismissible && (
            <ToastButtonCloseStyled onClick={onClose}>
              <CloseIcon />
            </ToastButtonCloseStyled>
          )}
          <ToastMessageStyled>{message}</ToastMessageStyled>
        </ToastTextContainer>
        <ToastActionsStyled hasChildren={Boolean(children)}>
          {children}
        </ToastActionsStyled>
      </ToastContentStyled>
    </ToastStyled>
  );
};

export default Toast;
