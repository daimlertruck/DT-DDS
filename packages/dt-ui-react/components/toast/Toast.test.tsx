import { fireEvent, render, screen } from '@testing-library/react';
import React from 'react';

import { withProviders } from '../../utils';

import { ToastType } from './constants';
import Toast from './Toast';

describe('<Toast /> component', () => {
  const TOAST_ID = 'myId';
  const TITLE = 'my title text';
  const MESSAGE = 'my message text';

  const onCloseFn = jest.fn();

  const ProvidedToast = withProviders(Toast);

  describe('when the toast type is success', () => {
    it('should render a toast with title & message', () => {
      const { container } = render(
        <ProvidedToast
          id={TOAST_ID}
          type={ToastType.Success}
          title={TITLE}
          message={MESSAGE}
          onClose={onCloseFn}
        />
      );

      expect(container).toMatchSnapshot();
    });

    it('should render a toast with title, message & children', () => {
      const { container } = render(
        <ProvidedToast
          id={TOAST_ID}
          type={ToastType.Success}
          title={TITLE}
          message={MESSAGE}
          onClose={onCloseFn}
        >
          Some children
        </ProvidedToast>
      );

      expect(container).toMatchSnapshot();
    });
  });

  describe('when the close button is clicked', () => {
    it('should fire the mock function', () => {
      render(
        <ProvidedToast
          id={TOAST_ID}
          type={ToastType.Success}
          title={TITLE}
          message={MESSAGE}
          onClose={onCloseFn}
        />
      );

      const closeBtn = screen.getByRole('button');
      fireEvent.click(closeBtn);

      expect(onCloseFn).toBeCalledTimes(1);
    });
  });
});
