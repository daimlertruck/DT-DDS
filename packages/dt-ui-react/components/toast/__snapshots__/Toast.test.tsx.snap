// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toast /> component when the toast type is success should render a toast with title & message 1`] = `
@keyframes animation-0 {
  from {
    opacity: 0;
  }

  to {
    opacity: 1;
  }
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 335px;
  -webkit-animation: animation-0 0.25s ease-in;
  animation: animation-0 0.25s ease-in;
  -webkit-transition: all 0.25s ease-in-out;
  transition: all 0.25s ease-in-out;
  background-color: #FFFFFF;
  opacity: 1;
  color: #151617;
}

.emotion-2 {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 8px;
  padding-right: 8px;
  background-color: #21A330;
  color: #FFFFFF;
}

.emotion-4 {
  height: 100%;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  padding: 12px 0;
  margin: 0 24px 0 16px;
}

.emotion-6 {
  text-transform: uppercase;
  letter-spacing: 0.5px;
  font-weight: 500;
  font-size: 1rem;
  line-height: 1.5rem;
  letter-spacing: normal;
  margin: 0;
  margin-bottom: 4px;
}

.emotion-8 {
  overflow: hidden;
  display: -webkit-box;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;
  font-weight: 400;
  font-size: 12px;
  line-height: 16.08px;
  letter-spacing: normal;
  margin: 0;
}

.emotion-10 {
  border: 0;
  cursor: pointer;
  background: transparent;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: flex-start;
  align-self: flex-start;
  margin-right: 12px;
  margin-top: 12px;
  border-radius: 3px;
  padding: 5px;
  font-size: 0;
  line-height: 0;
}

.emotion-10:hover {
  background-color: #636464;
}

<div>
  <div
    class="emotion-0 emotion-1"
    data-testid="toast-myId"
    type="success"
  >
    <div
      class="emotion-2 emotion-3"
      type="success"
    >
      <svg
        fill="currentColor"
        height="20"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10 0C4.48 0 0 4.48 0 10s4.48 10 10 10 10-4.48 10-10S15.52 0 10 0Zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8Zm4.59-12.42L8 12.17 5.41 9.59 4 11l4 4 8-8-1.41-1.42Z"
          fill="currentColor"
        />
      </svg>
    </div>
    <div
      class="emotion-4 emotion-5"
    >
      <div
        class="emotion-6 emotion-7"
      >
        my title text
      </div>
      <div
        class="emotion-8 emotion-9"
      >
        my message text
      </div>
    </div>
    <button
      class="emotion-10 emotion-11"
    >
      <svg
        fill="none"
        height="12"
        width="12"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m6.94 6 4.2-4.193a.67.67 0 0 0-.947-.947l-4.194 4.2L1.806.86a.67.67 0 1 0-.947.947L5.06 6l-4.2 4.194a.666.666 0 0 0 0 .946.667.667 0 0 0 .947 0l4.193-4.2 4.194 4.2a.668.668 0 1 0 .946-.947L6.94 6Z"
          fill="currentColor"
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`<Toast /> component when the toast type is success should render a toast with title, message & children 1`] = `
@keyframes animation-0 {
  from {
    opacity: 0;
  }

  to {
    opacity: 1;
  }
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 335px;
  -webkit-animation: animation-0 0.25s ease-in;
  animation: animation-0 0.25s ease-in;
  -webkit-transition: all 0.25s ease-in-out;
  transition: all 0.25s ease-in-out;
  background-color: #FFFFFF;
  opacity: 1;
  color: #151617;
}

.emotion-2 {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 8px;
  padding-right: 8px;
  background-color: #21A330;
  color: #FFFFFF;
}

.emotion-4 {
  height: 100%;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  padding: 12px 0;
  margin: 0 24px 0 16px;
}

.emotion-6 {
  text-transform: uppercase;
  letter-spacing: 0.5px;
  font-weight: 500;
  font-size: 1rem;
  line-height: 1.5rem;
  letter-spacing: normal;
  margin: 0;
  margin-bottom: 4px;
}

.emotion-8 {
  overflow: hidden;
  display: -webkit-box;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;
  font-weight: 400;
  font-size: 12px;
  line-height: 16.08px;
  letter-spacing: normal;
  margin: 0;
}

.emotion-10 {
  border: 0;
  cursor: pointer;
  background: transparent;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: flex-start;
  align-self: flex-start;
  margin-right: 12px;
  margin-top: 12px;
  border-radius: 3px;
  padding: 5px;
  font-size: 0;
  line-height: 0;
}

.emotion-10:hover {
  background-color: #636464;
}

<div>
  <div
    class="emotion-0 emotion-1"
    data-testid="toast-myId"
    type="success"
  >
    <div
      class="emotion-2 emotion-3"
      type="success"
    >
      <svg
        fill="currentColor"
        height="20"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10 0C4.48 0 0 4.48 0 10s4.48 10 10 10 10-4.48 10-10S15.52 0 10 0Zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8Zm4.59-12.42L8 12.17 5.41 9.59 4 11l4 4 8-8-1.41-1.42Z"
          fill="currentColor"
        />
      </svg>
    </div>
    <div
      class="emotion-4 emotion-5"
    >
      <div
        class="emotion-6 emotion-7"
      >
        my title text
      </div>
      <div
        class="emotion-8 emotion-9"
      >
        my message text
      </div>
      Some children
    </div>
    <button
      class="emotion-10 emotion-11"
    >
      <svg
        fill="none"
        height="12"
        width="12"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m6.94 6 4.2-4.193a.67.67 0 0 0-.947-.947l-4.194 4.2L1.806.86a.67.67 0 1 0-.947.947L5.06 6l-4.2 4.194a.666.666 0 0 0 0 .946.667.667 0 0 0 .947 0l4.193-4.2 4.194 4.2a.668.668 0 1 0 .946-.947L6.94 6Z"
          fill="currentColor"
        />
      </svg>
    </button>
  </div>
</div>
`;
